project_status_report/models/project_task_snapshot.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp (<http://www.camptocamp.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo import exceptions, _
+from odoo import SUPERUSER_ID
+
+from .common import TASK_FIELDS_TO_SYNC
+
+
+class ProjectTaskSnapshot(models.Model):
+    _name = 'project.task.snapshot'
+
+    report_id = fields.Many2one(comodel_name='project.status.report')
+    project_id = fields.Many2one(comodel_name='project.project',
+                                 related='report_id.project_id',
+                                 store=True,
+                                 readonly=True)
+    task_id = fields.Many2one(comodel_name='project.task',
+                              required=True,)
+    name = fields.Char(related='task_id.name', readonly=True)
+    user_id = fields.Many2one(comodel_name='res.users', string='User')
+    date_deadline = fields.Date()
+    progress = fields.Float()
+    planned_hours = fields.Float()
+    kanban_state = fields.Selection([('normal', 'In Progress'),
+                                     ('done', 'Ready for next stage'),
+                                     ('blocked', 'Blocked')],
+                                    string='Kanban State',
+                                    )
+    stage_id = fields.Many2one(comodel_name='project.task.type')
+
+    def _prepare_update_vals(self, task):
+        """
+        prepare dict value to make a snapshot of a task
+        """
+        res = {field: task[field]
+               for field, ftype in TASK_FIELDS_TO_SYNC.iteritems()
```

Comment:
what if in the future you change a field type?
Why not inspect the field type?
```python
fields_info = self.env[model].fields_get()
if fields_info[fname]['type'] == 'many2one':
[...]
```

Reply:
better yet, we can use read(fields, load=classic write) 

(says @gurneyalex )